utils/blockhash.mjs

```diff
@@ -94,5 +103,16 @@ let latestBlockhashPromise = new Promise((resolve) => {
 })();
 
 export async function getLatestBlockhash() {
-  return await latestBlockhashPromise;
+  const latestBlockhashes = await latestBlockhashesPromise;
+  const latestUnusedBlockhash = latestBlockhashes.find(
+    ({ blockhash }) => !usedBlockhashes.has(blockhash),
+  );
```

Comment:
This handles the _extremely_ rare case that you're:

1. Sending an identical transaction
2. Asked for a latest blockhash quickly enough to get the same one as last time (eg. maybe the blockhash fetcher has fallen behind)

Reply:
Wouldn't the `usedBlockhashes` list increase the memory consumption over time given the ever running nature of the ping thing?

Reply:
An identical transaction should be fine since, and, if transactions need not be identical then every time a blockhash is read from the watcher, it can have a value like `lastServedBlockhash` and if both the previously served blockhash and current one are same then handle it by waiting or polling for a new blockhash

Reply:
> Wouldn't the usedBlockhashes list increase the memory consumption over time given the ever running nature of the ping thing?

Nope! The code above copies over only the used hashes that are still valid. The maximum size of `usedBlockhashes` is the number of valid blockhashes.